.github/workflows/codeql.yml

```diff
@@ -0,0 +1,71 @@
+name: "CodeQL Advanced"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  schedule:
+    - cron: '38 10 * * 3'
+
+jobs:
+  analyze:
+    name: Analyze (${{ matrix.language }})
+    # Use Windows Server 2022 for Go analysis
+    runs-on: ${{ (matrix.language == 'go' && 'windows-2022') }}
+    permissions:
+      # required for all workflows
+      security-events: write
+
+      # required to fetch internal or private CodeQL packs
+      packages: read
+
+      # only required for workflows in private repositories
+      actions: read
+      contents: read
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - language: go
+            build-mode: autobuild
+        # Add other languages to the matrix as needed, e.g., 'c-cpp', 'python', etc.
+    steps:
+      # Checkout repository
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Initialize Go environment for Windows
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.22' # Adjust Go version as needed
+
+      # Cache Go modules to speed up builds
+      - name: Cache Go modules
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/.golangci-lint/cache
+          key: ${{ runner.os }}-go-modules-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-modules-
+
+      # Initialize CodeQL
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v3
+        with:
+          languages: ${{ matrix.language }}
+          build-mode: ${{ matrix.build-mode }}
+
+      - name: Verify Dependencies
```

Comment:
Is this required? it's already part of existing pipeline: https://github.com/microsoft/hnslib/blob/main/.github/workflows/ci.yml

Reply:
Removed dependency verification check